typegate/src/runtimes/s3.ts

```diff
@@ -111,6 +113,34 @@ export class S3Runtime extends Runtime {
         };
       }
 
+      if (name === "upload") {
+        const { bucket } = matData;
+        return async ({ file, path }) => {
+          const { key } = nativeResult(
+            await native.s3_upload(
+              this.client,
+              bucket as string,
+              path as string,
+              new Uint8Array(await (file as File).arrayBuffer()),
+            ),
+          );
+          return key;
+        };
+      }
+
+      if (name === "download_url") {
```

Comment:
We likely need to support two mats:
1. presigned with some more parameters (see official s3 doc)
2. canonical or classical url which might be used with or without the bucket name (e.g. only the s3 key/path)